File: packages/jexlang-java/src/main/java/io/github/vinaykumar0339/eval/errors/SyntaxErrorLocation.java
package io.github.vinaykumar0339.eval.errors;

public class SyntaxErrorLocation {
    private int line;
    private int column;
    private String offendingSymbol;

    public SyntaxErrorLocation(int line, int column, String offendingSymbol) {
        this.line = line;
        this.column = column;
        this.offendingSymbol = offendingSymbol;
    }

    public int getLine() {
        return line;
    }

    public void setLine(int line) {
        this.line = line;
    }

    public int getColumn() {
        return column;
    }

    public void setColumn(int column) {
        this.column = column;
    }

    public String getOffendingSymbol() {
        return offendingSymbol;
    }

    public void setOffendingSymbol(String offendingSymbol) {
        this.offendingSymbol = offendingSymbol;
    }

    @Override
    public String toString() {
        return "SyntaxErrorLocation{" +
                "line=" + line +
                ", column=" + column +
                ", offendingSymbol='" + offendingSymbol + '\'' +
                '}';
    }
}

File: packages/jexlang-swift/jexlang-swift/eval/errors/SyntaxErrorLocation.swift
//
//  SyntaxErrorLocation.swift
//  jexlang-swift
//
//  Created by Vinay Kumar on 03/10/25.
//

import Foundation

public struct SyntaxErrorLocation: CustomStringConvertible {
    private var line: Int
    private var column: Int
    private var offendingSymbol: String?
    
    public init(line: Int, column: Int, offendingSymbol: String? = nil) {
        self.line = line
        self.column = column
        self.offendingSymbol = offendingSymbol
    }
    
    public func getLine() -> Int {
        return line
    }
    
    public mutating func setLine(line: Int) {
        self.line = line
    }
    
    public func getColumn() -> Int {
        return column;
    }

    public mutating func setColumn(column: Int) {
        self.column = column;
    }

    public func getOffendingSymbol() -> String? {
        return offendingSymbol;
    }

    public mutating func setOffendingSymbol(offendingSymbol: String) {
        self.offendingSymbol = offendingSymbol;
    }
    
    public var description: String {
        return "SyntaxErrorLocation(line: \(line), column: \(column), offendingSymbol: \(String(describing: offendingSymbol)))"
    }
    
    
}
